### ISSUE TITLE
[Bug]: f-shortdoc.el:31:1:Error: Cannot open load file: No such file or directory, shortdoc

### ISSUE THREAD

**Issue by djpeebles** [rejeep/f.el#111]
### Expected behavior

Hi.  I've been getting an error when updating recently. I suspect that this is because shortdoc is only available for more recent versions of Emacs (I'm using version 27.2).

### Actual behavior

When updating over the last couple of weeks I get following output.

`Compiling internal form(s) at Wed Jun  8 07:35:09 2022
Leaving directory ‘/home/david/.emacs.d/elpa/f-20220607.1618’

Compiling file /home/david/.emacs.d/elpa/f-20220607.1618/f-shortdoc.el at Wed Jun  8 07:35:09 2022
Entering directory ‘/home/david/.emacs.d/elpa/f-20220607.1618/’
f-shortdoc.el:31:1:Error: Cannot open load file: No such file or directory, shortdoc

Compiling file /home/david/.emacs.d/elpa/f-20220607.1618/f.el at Wed Jun  8 07:35:09 2022`

### f.el version

master

### Emacs version

27.2

**Comment by Phundrak** [comment]
That’s weird, `f-shortdoc.el' should only be required if your Emacs version is at least 28.1. I’ll push a hotfix for that, but I’ll keep this issue open while looking for a more elegant solution.

--
Lucien “Phundrak” Cartier-Tilet
<https://phundrak.com> (Français)
<https://phundrak.com/en> (English)
Sent from GNU/Emacs

**Comment by Dickby** [comment]
This errors on emacs 29.05 with `Debugger entered--Lisp error: (void-variable f)`

**Comment by djpeebles** [comment]
> That’s weird, `f-shortdoc.el' should only be required if your Emacs version is at least 28.1. I’ll push a hotfix for that, but I’ll keep this issue open while looking for a more elegant solution.

Thank you!

**Comment by Phundrak** [comment]
I attempted to restore shortdoc, this should fix the initial bug in this issue (the second bug report was due to its hotfix). Let me know if there is anything wrong happening again with the shortdoc

**Comment by Dickby** [comment]
Works! 
Thanks a lot

**Comment by djpeebles** [comment]
Yes, I've just updated the package too and it now works without any errors.  Thank you!

**Comment by Phundrak** [comment]
Perfect, closing this issue

**Comment by mittenchops** [comment]
I'm getting this error in emacs 29.0.5.

**Comment by Phundrak** [comment]
I reverted commit https://github.com/rejeep/f.el/commit/6544f84aa5adaa76ea8901f185e940ee868a471f which should fix the issue.

**Comment by onetom** [comment]
Just for the record, I'm getting this error on `GNU Emacs 30.0.50 (build 1, aarch64-apple-darwin22.4.0, NS appkit-2299.50 Version 13.3.1 (a) (Build 22E772610a)) of 2023-05-19` too, when I'm trying to run the `helpful-key` command from the https://github.com/wilfred/helpful package.

I don't understand based on the conversation above what am I supposed to do.
I'm using `straight.el` and I did a `straight-pull-all` and a restart, but that didn't solve the problem either, so I've just disabled the `helpful` package.
